dashboard/app/models/course_version.rb

```diff
@@ -18,4 +18,31 @@
 
 class CourseVersion < ApplicationRecord
   belongs_to :content_root, polymorphic: true
+
+  def self.update_course_version(content_root)
+    # TODO: Once the Course model is added, change this to just be version_year, since then the
+    # unique index will be on (course_id, key).
+    key = "#{content_root.family_name}-#{content_root.version_year}"
+
+    unless content_root.is_course?
```

Comment:
Is this here if someone were to change is_course from true to false? Maybe a comment explaining that would be helpful.  Also again wondering does do the `content_root.course_version =` delete the old object and if so could we just set that to nil for this case?

Reply:
Maybe it is just too early in the morning (😆 ) but I think I am getting confused about what is_course means. In a script file, it made sense to me as "this is a script which represents an entire course". what would it mean for `unit_group.is_course?` to be true?

Separately, I don't see where `unit_group.is_course?` would be defined so it looks like this might not be working when the content root is a unit group. Am I missing something?

Reply:
Yea, I haven't added that case yet (listed under future work). I think the implementation there should just be adding an `is_course` method to the UnitGroup class which always returns true, since AFAIK every unit group we have right now corresponds to a top level CourseVersion.

Do you think this makes sense? Is there a clearer way to do this?

Reply:
I think that makes sense, but in that case i think what's missing here is an "interface" defining the methods and contracts that unit_group and script are both required to implement in order to be able to be a content root. I can't remember what we found earlier about whether this is possible to accomplish in ruby, but if it's not possible to truly implement an interface then some comments on the content_root association describing the interface could be an acceptable substitute.

Reply:
I added comments to document the "Interface" as it is so far in course_version.rb. I can explore the unit test based "interface" enforcement when I work on adding seeding for .course files.